relayer/chain/parachain/writer.go

```diff
@@ -80,13 +80,13 @@ func (wr *ParachainWriter) BatchCall(ctx context.Context, extrinsic string, call
 		slicedCalls := append([]interface{}{}, calls[i:j]...)
 		encodedCalls := make([]types.Call, len(slicedCalls))
 		for k := range slicedCalls {
-			call, err := wr.prepCall(extrinsic, slicedCalls[k])
+			call, err := wr.prepCall(extrinsic[k], slicedCalls[k])
 			if err != nil {
 				return err
 			}
 			encodedCalls[k] = *call
 		}
-		err := wr.WriteToParachainAndRateLimit(ctx, "Utility.batch_all", encodedCalls)
+		err := wr.WriteToParachainAndWatch(ctx, "Utility.batch_all", encodedCalls)
```

Comment:
Will this call get blocked?

Reply:
Yes, it will wait until the `batchall` request has been completed and it is a block.